src/futures.coffee

```diff
@@ -85,3 +85,27 @@ Future.handle = (prom, fn) ->
   prom.fail (args...) ->
     deferred.reject(fn.apply(null, args))
   Future deferred.promise()
+
+
+Future.retry = (futureClosure, backoffGenerator) ->
+  successXHR = $.Deferred()
+  retryingFunc = ->
+    futureClosure().done(->
+        successXHR.resolve.apply(successXHR, arguments)
+      ).fail(->
+        backoffValue = backoffGenerator()
+        if backoffValue == null
+          successXHR.reject.apply(successXHR, arguments)
+        else
+          setTimeout(retryingFunc, backoffValue)
+      )
+  retryingFunc()
+  successXHR
+
+
+Future.retryWithConstantBackoff = (futureClosure, interval, maxAttempts) ->
```

Comment:
I like the maxAttempts interface for retries.  I've seen it as "retries" before, where "retries" maybe means "maxAttempts - 1" and maybe means maxAttempts.